newm-server/src/main/kotlin/io/newm/server/ext/ApplicationCallExt.kt

```diff
@@ -52,3 +60,37 @@ val ApplicationCall.olderThan: LocalDateTime?
 
 val ApplicationCall.newerThan: LocalDateTime?
     get() = parameters["newerThan"]?.toLocalDateTime()
+
+/***
+ * Receives signed content and verifies signature
+ */
+suspend inline fun <reified T : Any> ApplicationCall.receiveAndVerify(
+    signatureHeader: String,
+    key: Key
+): T = receiveAndVerify(signatureHeader, key, T::class)
+
+/***
+ * Receives signed content and verifies signature
+ */
+suspend fun <T : Any> ApplicationCall.receiveAndVerify(
+    signatureHeader: String,
+    key: Key,
+    type: KClass<T>
+): T {
+    val signature = request.headers[signatureHeader] ?: throw HttpUnauthorizedException("missing signature")
+    val mac = Mac.getInstance(key.algorithm).apply {
+        init(key)
+    }
+    request.pipeline.intercept(ApplicationReceivePipeline.Before) { data ->
+        val channel = data as ByteReadChannel
+        val size = channel.availableForRead
+        val buffer = ByteBuffer.allocate(size)
```

Comment:
Add a new ticket to improve this in the future. I believe you can use the `ByteBufferPool` so you're not continually allocating a new buffer with every request.

Reply:
Will do.